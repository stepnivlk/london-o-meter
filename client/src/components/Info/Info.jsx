// @flow

import React from 'react';

const Info = () => (
  <div className="Info">
    <h3>What kind of magic is this?</h3>
    <p>
      This is an experiment with Twitter data mining and its analysis. Please don't take the output too seriously.
    </p>
    <p>
      Basically, there's an IFTTT applet triggering a web hook action which sends all the tweets in London to the particular Webtask URL.
      Webtask then serializes the tweet data and stores it in an ElasticSearch index.
    </p>
    <p>
      Then there's a scheduled Webtask action which processes batches of tweets from an ElasticSearch, contacts text analyzation service and updates data in ElasticSearch with its output.
    </p>
    <p>
      Finally, there's a Webtask acting as an API for this React + D3.js client which aggregates data from ElasticSearch and serves it as a JSON.
    </p>
    <h4>Technologies used:</h4>
    <p><a href='http://ifttt.com/' target='_blank' rel='noopener noreferrer'>IFTTT</a></p>
    <p><a href='https://webtask.io/' target='_blank' rel='noopener noreferrer'>Webtask</a></p>
    <p><a href='https://www.elastic.co/' target='_blank' rel='noopener noreferrer'>ElasticSearch</a></p>
    <p><a href='https://uclassify.com/' target='_blank' rel='noopener noreferrer'>uClassify</a></p>
    <p><a href='https://facebook.github.io/react/' target='_blank' rel='noopener noreferrer'>React</a></p>
    <p><a href='https://d3js.org/' target='_blank' rel='noopener noreferrer'>D3.js</a></p>
    <p>[ <a href='https://github.com/stepnivlk/' target='_blank' rel='noopener noreferrer'>author</a> ][ <a href='https://github.com/stepnivlk/london-o-meter' target='_blank' rel='noopener noreferrer'>code</a> ]</p>
  </div>
);

export default Info;
