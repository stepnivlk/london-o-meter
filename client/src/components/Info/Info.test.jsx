import React from 'react';
import renderer from 'react-test-renderer';
import Info from './Info';

it('matches a snapshot', () => {
  const component = renderer.create(
    <Info />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
