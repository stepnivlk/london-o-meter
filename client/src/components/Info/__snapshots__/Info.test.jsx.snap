// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches a snapshot 1`] = `
<div
  className="Info"
>
  <h3>
    What kind of magic is this?
  </h3>
  <p>
    This is an experiment with Twitter data mining and its analysis. Please don't take the output too seriously.
  </p>
  <p>
    Basically, there's an IFTTT applet triggering a web hook action which sends all the tweets in London to the particular Webtask URL. Webtask then serializes the tweet data and stores it in an ElasticSearch index.
  </p>
  <p>
    Then there's a scheduled Webtask action which processes batches of tweets from an ElasticSearch, contacts text analyzation service and updates data in ElasticSearch with its output.
  </p>
  <p>
    Finally, there's a Webtask acting as an API for this React + D3.js client which aggregates data from ElasticSearch and serves it as a JSON.
  </p>
  <h4>
    Technologies used:
  </h4>
  <p>
    <a
      href="http://ifttt.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      IFTTT
    </a>
  </p>
  <p>
    <a
      href="https://webtask.io/"
      rel="noopener noreferrer"
      target="_blank"
    >
      Webtask
    </a>
  </p>
  <p>
    <a
      href="https://www.elastic.co/"
      rel="noopener noreferrer"
      target="_blank"
    >
      ElasticSearch
    </a>
  </p>
  <p>
    <a
      href="https://uclassify.com/"
      rel="noopener noreferrer"
      target="_blank"
    >
      uClassify
    </a>
  </p>
  <p>
    <a
      href="https://facebook.github.io/react/"
      rel="noopener noreferrer"
      target="_blank"
    >
      React
    </a>
  </p>
  <p>
    <a
      href="https://d3js.org/"
      rel="noopener noreferrer"
      target="_blank"
    >
      D3.js
    </a>
  </p>
  <br />
  <p>
    [ 
    <a
      href="https://github.com/stepnivlk/"
      rel="noopener noreferrer"
      target="_blank"
    >
      author
    </a>
     ][ 
    <a
      href="https://github.com/stepnivlk/london-o-meter"
      rel="noopener noreferrer"
      target="_blank"
    >
      code
    </a>
     ]
  </p>
</div>
`;
